### DISCUSSION TITLE
Using spy, why does event type action with object undefined fire after event type reaction?

### DISCUSSION THREAD

**Discussion by ghost** [mobxjs/mobx#2958] [Q&A] [answered]
While logging the `spy` events I noticed that once I added a reaction to the store, every time an action ran that changed the observable used in the reaction, two events got logged. 1st the reaction and 2nd an action with `.object: undefined`.

I've created a [reproducible sandbox](https://codesandbox.io/s/staging-pond-krf6t). If you select the other drop down entry you'll see `REACTION Reaction!` and `ACTION  undefined.Reaction!()` afterwards in the console.

Why does it happen? Is the 1st reaction from "data function" and 2nd action from "side effect" function?

**Comment by urugator** [comment, marked as answer]
> 2nd action from "side effect" function?

Yes, the effect function is implicitely wrapped in standard `action`
https://github.com/mobxjs/mobx/blob/eb4fb5d61403aecbbb2546dd154a2e5431128ce5/packages/mobx/src/api/autorun.ts#L115-L118

**Reply by ghost** [reply]
Thank you for the quick answer and for providing the exact lines where it's defined!
